temporal-serviceclient/src/main/java/io/temporal/internal/retryer/GrpcRetryerUtils.java

```diff
@@ -56,6 +53,15 @@ class GrpcRetryerUtils {
         // https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/retry/retry.go#L287
       case CANCELLED:
         return new CancellationException("The gRPC request was cancelled");
+      case INVALID_ARGUMENT:
+      case NOT_FOUND:
+      case ALREADY_EXISTS:
+      case FAILED_PRECONDITION:
+      case PERMISSION_DENIED:
+      case UNAUTHENTICATED:
+      case UNIMPLEMENTED:
+        // never retry these codes
+        return currentException;
```

Comment:
Here's what Go retries: https://github.com/temporalio/sdk-go/blob/v1.14.0/internal/common/retry/interceptor.go#L119-L124. It looks like it matches up with this.